runtime/permissions/lib.rs

```diff
@@ -869,138 +1069,317 @@ impl AsRef<str> for EnvDescriptor {
 }
 
 #[derive(Clone, Eq, PartialEq, Hash, Debug, Serialize, Deserialize)]
-pub struct RunPathQuery<'a> {
-  pub requested: &'a str,
-  pub resolved: &'a Path,
-}
-
-pub enum RunDescriptorArg {
+pub enum RunQueryDescriptor {
+  Path {
+    requested: String,
+    resolved: PathBuf,
+  },
+  /// This variant can't be used to grant permissions. It's mostly
+  /// used so that prompts and everything works the same way as when
+  /// the command is resolved, meaning that a script can't tell
+  /// if a command is resolved or not based on how long something
+  /// takes to ask for permissions.
```

Comment:
Could you reword this docstring? I'm having trouble understanding it - what does "This variant can't be used to grant permissions" mean?

Reply:
I updated it a bit. The variant gets created when the command path can't be resolved, so it will prompt the users for permissions about it, but it doesn't actually do anything when the user grants permissions. It's to prevent someone trying to probe for permissions to find out what's on the path.